l1-contracts/contracts/zksync/libraries/BLS.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+library BLS {
+    // Field order
+    uint256 private constant N = 21888242871839275222246405745257275088696311157297823662689037894645226208583;
+
+    // (N + 1) / 4
+    uint256 private constant N14 = 5472060717959818805561601436314318772174077789324455915672259473661306552146;
+
+    // Negated generator of G2
+    uint256 private constant N_G2_X1 = 11559732032986387107991004021392285783925812861821192530917403151452391805634;
+    uint256 private constant N_G2_X0 = 10857046999023057135944570762232829481370756359578518086990519993285655852781;
+    uint256 private constant N_G2_Y1 = 17805874995975841540914202342111839520379459829704422454583296818431106115052;
+    uint256 private constant N_G2_Y0 = 13392588948715843804641432497768002650278120570034223513918757245338268106653;
+
+    // Generator of G1
+    uint256 private constant G1_X = 7875458235035678754887153468411793526875066621955642619646139314277366414792;
+    uint256 private constant G1_Y = 8106623690154677659962327366078301943507317780154727084061147098569974335996;
+
+    function verifyAggregation(
+        uint256[2][] calldata pk1s,
+        uint256[4] calldata aggregated_pk2s,
+        uint256[2] calldata aggregated_sig,
+        bytes calldata msg,
+        uint256 nonce
+    ) internal view returns (bool) {
+        require(pk1s.length >= 2, "BLS: number of pk1s must be at least 2");
+
+        uint256[2] memory message_point = hashToPoint(msg, nonce);
+        if (!ecPairing(aggregated_sig, aggregated_pk2s, message_point)) {
```

Comment:
Please document why we do not need a subgroup check for the signature. It will allow to decrease the number of FP in audit contests